src/main/java/com/courtcanva/ccfranchise/exceptions/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.courtcanva.ccfranchise.exceptions;
+
+import com.courtcanva.ccfranchise.dtos.ErrorDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ErrorDto handleResourceNotFoundException(ResourceNotFoundException e) {
+
+        log.error(e.getMessage(), e);
```

Comment:
We should have an explicit log for each exception instead of just documenting the exception message. Besides, we only need to put `e`, at the end of the argument list to document the error stack trace,  `e.getMessage` is not needed. Other than that, given this exception is customized and thrown by us, I recommend adding log to where we throw it to make the error message more specific.

Reply:
thx, Lao. I already add a specific description for the log